src/api/web/endpoint/package.ts

```diff
@@ -119,8 +118,7 @@ function addPackageWebApi(route: Router, storage: IStorageHandler, auth: IAuth,
           const {v} = req.query;
           let sideBarInfo: any = _.clone(info);
           sideBarInfo.versions = convertDistRemoteToLocalTarballUrls(info, req, config.url_prefix).versions;
-          if (isVersionValid(info, v)) {
-            // @ts-ignore
+          if (typeof v === 'string' && isVersionValid(info, v)) {
```

Comment:
I just noticed that `isVersionValid` already check for the type and seems redundant, you might replace the `undefined` check by the `string` and have the whole check within the method.